common/beerocks/bcl/include/bcl/beerocks_socket_thread.h

```diff
@@ -97,9 +97,10 @@ class socket_thread : public thread_base {
     int server_max_connections;
     SocketSelect select;
 
-    uint32_t m_select_timeout_msec                              = 0;
-    bool m_skip_next_select_timeout                             = false;
-    std::chrono::steady_clock::time_point m_select_wake_up_time = std::chrono::steady_clock::now();
+    uint32_t m_select_timeout_msec  = 0;
+    bool m_skip_next_select_timeout = false;
+    std::chrono::steady_clock::time_point m_select_wake_up_time =
+        std::chrono::steady_clock::time_point::min();
```

Comment:
I think this would be better:
```suggestion
        std::chrono::steady_clock::time_point::max();
```
Indeed, what you want to avoid is for this condition to trigger:
```
    auto awake_time_msec = std::chrono::duration_cast<std::chrono::milliseconds>(
                               std::chrono::steady_clock::now() - m_select_wake_up_time)
                               .count();

    if (m_select_timeout_msec > 0 && awake_time_msec > m_select_timeout_msec)
```
When you initialize it with min(), awake_time_msec should actually be very large (and positive, since now is definitely later than min) so larger than m_select_timeout_msec. The only reason that it happens to work is that it is so large that it overflows and thus becomes negative. But you can hardly rely on that... With max, it simply be negative and the condition won't trigger.

(My analysis could be wrong though.)

Reply:
Apparently both give the same value: https://repl.it/repls/BriefPoorKeygen
But no problem, I'll change it to max.